src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs

```diff
@@ -322,7 +321,7 @@ protected Next GetNextActionForPage(ApplicationSection section, Data.Entities.Ap
                         if (!String.IsNullOrWhiteSpace(condition.QuestionTag))
                         {
                             var questionTagValue = applicationData[condition.QuestionTag];
-                            var questionTag = questionTagValue?.Value<string>();
+                            var questionTag = questionTagValue?.GetValue<string>();
```

Comment:
Remember how `applicationData["Q1"]` was non-null in Newtonsoft but null in Text.Json? I think we need to account for that behaviour here, otherwise `questionTagValue` will most likely end up being null when it shouldn't.

I think we should revert this to use Newtonsoft, refactor some of this code into a class, and add some tests to snapshot the old behaviour. Then change it to use Text.Json, re-run the tests, and see what needs to be changed to account for the new behaviour.

This one might be a bit messy, so let me know if you need help.

Reply:
You've pointed out that there are no null checks -- resolving